### PR TITLE
Maypad improvements

### PR DESCRIPTION
**Pull request by eosti**
## Description

Adds a splitPlus layout option to the Maypad keyboard. Good option if you want those nice large Enter and 0 keys but really can't see yourself using the plus key too much. 
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keyhive/maypad/rules.mk [hunk 1]
```diff
@@ -32,4 +32,4 @@ AUDIO_ENABLE = no           # Audio output on port C6
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
 HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs
 
-LAYOUTS = ortho_5x4 numpad_5x4
+LAYOUTS = ortho_5x4 numpad_5x4 numpad_5x4_splitPlus
```
Comment (zvecr) [inline keyboards/keyhive/maypad/rules.mk:35]:
```suggestion
LAYOUTS = ortho_5x4 numpad_5x4
```

File: keyboards/keyhive/maypad/readme.md [hunk 2]
```diff
@@ -10,4 +10,9 @@ Make example for this keyboard (after setting up your build environment):
 
     make keyhive/maypad:default
 
-See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
\ No newline at end of file
+There are three layouts, depending on how you assembled your Maypad:
+* `LAYOUT`, a 5x4 ortho layout
+* `LAYOUT_numpad`, a full numpad layout with 2U 0, Enter, and + keys
+* `LAYOUT_numpad_splitPlus`, a full numpad layout with only 2U 0 and Enter keys
+
```
Comment (zvecr) [inline keyboards/keyhive/maypad/readme.md:17]: Personally i don't think this is needed.
```suggestion
```

File: keyboards/keyhive/maypad/maypad.h [hunk 3]
```diff
@@ -59,3 +73,7 @@
     { k30, k31, k32, k33 }, \
     { k40, k41, k42, k43 }  \
 }
+
+#define LAYOUT                  LAYOUT_ortho_5x4
+#define LAYOUT_numpad           LAYOUT_numpad_5x4
+#define LAYOUT_numpad_splitPlus LAYOUT_numpad_5x4_splitPlus
```
Comment (zvecr) [inline keyboards/keyhive/maypad/maypad.h:79]:
```suggestion
```

File: keyboards/keyhive/maypad/info.json [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
+    "keyboard_name": "maypad",
+    "url": "https://keyhive.xyz/shop/may-pad",
+    "maintainer": "codybender",
+    "width": 4,
+    "height": 5,
     "layouts": {
         "LAYOUT_numpad_5x4": {
             "layout": [{"label":"Num Lock", "x":0, "y":0}, {"label":"/", "x":1, "y":0}, {"label":"*", "x":2, "y":0}, {"label":"-", "x":3, "y":0}, {"label":"7", "x":0, "y":1}, {"label":"8", "x":1, "y":1}, {"label":"9", "x":2, "y":1}, {"label":"4", "x":0, "y":2}, {"label":"5", "x":1, "y":2}, {"label":"6", "x":2, "y":2}, {"label":"+", "x":3, "y":1, "h":2}, {"label":"1", "x":0, "y":3}, {"label":"2", "x":1, "y":3}, {"label":"3", "x":2, "y":3}, {"label":"0", "x":0, "y":4, "w":2}, {"label":".", "x":2, "y":4}, {"label":"Enter", "x":3, "y":3, "h":2}]
         },
+        "LAYOUT_numpad_5x4_splitPlus": {
```
Comment (zvecr) [inline keyboards/keyhive/maypad/info.json:11]:
```suggestion
        "LAYOUT_numpad_5x4_split_plus": {
```

File: keyboards/keyhive/maypad/maypad.h [hunk 5]
```diff
@@ -46,6 +46,20 @@
     { KC_NO, k41, k42, k43   }   \
 }
 
+#define LAYOUT_numpad_5x4_splitPlus( \
```
Comment (zvecr) [inline keyboards/keyhive/maypad/maypad.h:49]:
```suggestion
#define LAYOUT_numpad_5x4_split_plus( \
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Though overall, my suggestion would be to just use `LAYOUT_ortho_5x4` and double up any required keys.

File: keyboards/keyhive/maypad/rules.mk [hunk 6]
```diff
@@ -32,4 +32,4 @@ AUDIO_ENABLE = no           # Audio output on port C6
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
 HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs
 
-LAYOUTS = ortho_5x4 numpad_5x4
+LAYOUTS = ortho_5x4 numpad_5x4 numpad_5x4_split_plus
```
Comment (zvecr) [inline keyboards/keyhive/maypad/rules.mk:35]: This has to be list of valid community layouts
```suggestion
LAYOUTS = ortho_5x4 numpad_5x4
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/keyhive/maypad/rules.mk (hunk 6 above)
Comment (eosti) [reply keyboards/keyhive/maypad/rules.mk]: Ah, got it, I didn't realize that needed to be the case. Thanks for the help!

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**